src/components/TodoModal/TodoModal.tsx

```diff
@@ -1,49 +1,80 @@
-import React from 'react';
+import { FC, useEffect, useState } from 'react';
+import { getUser } from '../../api';
+import { OptionForFilterTodos } from '../../types/OptionForFilterTodos';
+import { Todo } from '../../types/Todo';
+import { User } from '../../types/User';
 import { Loader } from '../Loader';
 
-export const TodoModal: React.FC = () => {
+interface Props {
+  selectedTodo: Todo,
+  setSelectedTodoId: (selectedTodoId: number | null) => void,
+  setOptionForFilter: (optionForFilter: OptionForFilterTodos) => void,
+}
+
+export const TodoModal: FC<Props> = (props) => {
+  const { selectedTodo, setSelectedTodoId, setOptionForFilter } = props;
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [isError, setIsError] = useState(false);
+
+  useEffect(() => {
+    setIsError(false);
+
+    getUser(selectedTodo.userId)
+      .then(setSelectedUser)
+      .catch(() => setIsError(true));
+  }, [selectedTodo]);
+
   return (
     <div className="modal is-active" data-cy="modal">
       <div className="modal-background" />
 
-      {true ? (
-        <Loader />
-      ) : (
-        <div className="modal-card">
-          <header className="modal-card-head">
-            <div
-              className="modal-card-title has-text-weight-medium"
-              data-cy="modal-header"
-            >
-              Todo #2
-            </div>
+      {!selectedUser
+        ? <Loader />
+        : (
+          <div className="modal-card">
+            <header className="modal-card-head">
+              <div
+                className="modal-card-title has-text-weight-medium"
+                data-cy="modal-header"
+              >
+                {`Todo #${selectedTodo.id}`}
+              </div>
 
-            {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-            <button
-              type="button"
-              className="delete"
-              data-cy="modal-close"
-            />
-          </header>
-
-          <div className="modal-card-body">
-            <p className="block" data-cy="modal-title">
-              quis ut nam facilis et officia qui
-            </p>
-
-            <p className="block" data-cy="modal-user">
-              {/* <strong className="has-text-success">Done</strong> */}
-              <strong className="has-text-danger">Planned</strong>
-
-              {' by '}
-
-              <a href="mailto:Sincere@april.biz">
-                Leanne Graham
-              </a>
-            </p>
+              {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
+              <button
+                type="button"
+                className="delete"
+                data-cy="modal-close"
+                onClick={() => {
+                  setSelectedTodoId(null);
+                  setOptionForFilter(OptionForFilterTodos.All);
+                }}
```

Comment:
I dont think you have to reset Filtering option here.
```suggestion
                data-cy="modal-close"
                onClick={() => {
                  setSelectedTodoId(null);
                }}
```